lib/src/migrators/division.dart

```diff
@@ -330,110 +366,81 @@ class _DivisionMigrationVisitor extends MigrationVisitor {
-  bool _operatesOnNumbers(BinaryOperator operator) =>
-      _returnsNumbers(operator) ||
-      operator == BinaryOperator.lessThan ||
-      operator == BinaryOperator.lessThanOrEquals ||
-      operator == BinaryOperator.greaterThan ||
-      operator == BinaryOperator.greaterThanOrEquals;
+  bool _operatesOnNumbers(BinaryOperator operator) => {
+        BinaryOperator.times,
+        BinaryOperator.modulo,
+        BinaryOperator.lessThan,
+        BinaryOperator.lessThanOrEquals,
+        BinaryOperator.greaterThan,
+        BinaryOperator.greaterThanOrEquals
+      }.contains(operator);
 
   /// Returns true if [node] is entirely composed of number literals and slash
   /// operations.
   bool _onlySlash(Expression node) {
-    if (node is NumberExpression) return true;
-    if (node is BinaryOperationExpression) {
-      return node.operator == BinaryOperator.dividedBy &&
-          _onlySlash(node.left) &&
-          _onlySlash(node.right);
+    switch (node) {
+      case NumberExpression():
+        return true;
+      case BinaryOperationExpression(
+          operator: BinaryOperator.dividedBy,
+          :var left,
+          :var right
+        ):
+        return _onlySlash(left) && _onlySlash(right);
+      default:
+        return false;
     }
-    return false;
-  }
-
-  /// Returns true if [node] is known to always evaluate to a number.
-  bool _isDefinitelyNumber(Expression node) {
-    if (node is NumberExpression) return true;
-    if (node is ParenthesizedExpression) {
-      return _isDefinitelyNumber(node.expression);
-    } else if (node is UnaryOperationExpression) {
-      return _isDefinitelyNumber(node.operand);
-    } else if (node is BinaryOperationExpression) {
-      return _returnsNumbers(node.operator) ||
-          (_isDefinitelyNumber(node.left) && _isDefinitelyNumber(node.right));
-    }
-    return false;
-  }
-
-  /// Returns true if [node] contains a subexpression known to not be a number.
-  bool _isDefinitelyNotNumber(Expression node) {
-    if (node is ParenthesizedExpression) {
-      return _isDefinitelyNotNumber(node.expression);
-    }
-    if (node is BinaryOperationExpression) {
-      return _isDefinitelyNotNumber(node.left) ||
-          _isDefinitelyNotNumber(node.right);
-    }
-    return node is BooleanExpression ||
-        node is ColorExpression ||
-        node is ListExpression ||
-        node is MapExpression ||
-        node is NullExpression ||
-        node is StringExpression;
   }
 
   /// Returns true if [node] contains an interpolation.
   bool _containsInterpolation(Expression node) {
-    if (node is ParenthesizedExpression) {
-      return _containsInterpolation(node.expression);
-    }
-    if (node is BinaryOperationExpression) {
-      return _containsInterpolation(node.left) ||
-          _containsInterpolation(node.right);
+    switch (node) {
+      case ParenthesizedExpression(:var expression):
+      case UnaryOperationExpression(operand: var expression):
```

Comment:
I generally try to use `||` rather than cascading `case`s because it's easier to translate to and from expression form

Reply:
Done